edgelet/edgelet-hsm/src/crypto.rs

```diff
@@ -29,36 +29,36 @@ pub use crate::error::{Error, ErrorKind};
 /// Activate a private key, and then you can use that key to sign data.
 #[derive(Clone)]
 pub struct Crypto {
-    crypto: Arc<Mutex<HsmCrypto>>,
+    crypto: Arc<HsmCrypto>,
+    hsm_lock: Arc<Mutex<()>>,
```

Comment:
I suggest introducing a newtype `pub struct HsmLock(Mutex<()>);`. It makes it clear what the purpose of the mutex is to users outside `edgelet_hsm`, and gives a hint that they shouldn't create a new `Mutex<()>` instead of reusing an existing one.

Reply:
Thanks, for your suggestion and help. Please have a look.